pkg/registry/backgroundsvcs/adapter/service.go

```diff
@@ -0,0 +1,63 @@
+package adapter
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/grafana/pkg/registry"
+)
+
+var _ services.NamedService = &serviceAdapter{}
+
+// serviceAdapter adapts a Grafana background service to dskit's NamedService interface.
+// It wraps the background service with dskit's BasicService to provide the standard
+// service state model: New → Starting → Running → Stopping → Terminated/Failed.
+//
+// The adapter uses dskit's BasicService with custom StartingFn and RunningFn:
+//   - Starting phase: No-op, transitions immediately to Running
+//   - Running phase: Delegates to the wrapped service's Run method
+//   - Stopping phase: Handled automatically by context cancellation
+type serviceAdapter struct {
+	*services.BasicService
+	name    string
+	service registry.BackgroundService
+}
+
+// asNamedService converts a Grafana background service into a dskit NamedService.
+// The returned service starts in the New state and can be managed using dskit's
+// standard service operations (StartAsync, AwaitRunning, StopAsync, AwaitTerminated).
+//
+// The service name is derived from the Go type name using reflection, ensuring
+// each service type has a unique identifier within the dskit module system.
+func asNamedService(service registry.BackgroundService) *serviceAdapter {
+	name := reflect.TypeOf(service).String()
+	a := &serviceAdapter{
+		name:    name,
+		service: service,
+	}
+	a.BasicService = services.NewBasicService(a.start, a.run, nil).WithName(name)
+	return a
+}
+
+// start implements the StartingFn for dskit's BasicService.
+// This phase transitions the service from New to Starting state, then immediately
+// to Running state since no initialization is required. The actual service work
+// happens in the run phase.
+func (a *serviceAdapter) start(ctx context.Context) error {
+	return nil
```

Comment:
`Starting` state is there for service to initialize itself. If there is nothing to do in Starting state, we can as well pass `nil` to `NewBasicService` and don't need an empty function.

Reply:
removed, thanks!